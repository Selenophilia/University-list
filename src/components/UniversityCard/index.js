import React from "react";
import Proptypes from "prop-types";
import { makeStyles } from "@material-ui/core/styles";
import {
  Button,
  Card,
  CardActionArea,
  CardActions,
  CardContent,
  CardMedia,
  Typography
} from "@material-ui/core";

const useStyles = makeStyles({
  root: {
    width: 350,
    marginBottom: 20,
    marginRight: 20,
    flex: "20% 0 0"
  },
  media: {
    height: 140
  },
  cardBody: {
    height: 80,
    fontSize: 22,
    lineHeight: 1.5,
    marginBottom: 20
  },
  cardFooter: {
    paddingLeft: 15,
    paddingRight: 15,
    display: "flex",
    justifyContent: "space-between"
  }
});

const UniversityCard = ({ name, country, image, page }) => {
  const classes = useStyles();
  const handleClick = (page) => {
    window.location.href = `${page}`;
  };
  return (
    <Card className={classes.root}>
      <CardActionArea>
        <CardMedia
          className={classes.media}
          image={image}
          title="Contemplative Reptile"
        />
        <CardContent>
          <Typography
            gutterBottom
            variant="h5"
            component="h2"
            className={classes.cardBody}
          >
            {name}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            {`Country: ${country}`}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions className={classes.cardFooter}>
        <Button size="small" color="primary">
          Share
        </Button>
        <Button size="small" color="primary" onClick={() => handleClick(page)}>
          Visit Website
        </Button>
      </CardActions>
    </Card>
  );
};

UniversityCard.propTypes = {
  name: Proptypes.string.isRequired,
  country: Proptypes.string.isRequired,
  image: Proptypes.string.isRequired,
  page: Proptypes.string.isRequired
};
export default UniversityCard;
