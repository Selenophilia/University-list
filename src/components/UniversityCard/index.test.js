import React from "react";
import { render } from "@testing-library/react";
import UniversityCard from ".";

const renderRoot = () => {
  return render(
    <UniversityCard
      name="College of university"
      country="Phillipines"
      image="https://source.unsplash.com/800x900/?university"
      page="google.com"
    />
  );
};

it("renders Footer component correctly ", () => {
  const { asFragment } = renderRoot();
  expect(asFragment()).toMatchSnapshot();
});
