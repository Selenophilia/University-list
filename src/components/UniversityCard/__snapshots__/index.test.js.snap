// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer component correctly  1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardMedia-root makeStyles-media-2"
        style="background-image: url(https://source.unsplash.com/800x900/?university);"
        title="Contemplative Reptile"
      />
      <div
        class="MuiCardContent-root"
      >
        <h2
          class="MuiTypography-root makeStyles-cardBody-3 MuiTypography-h5 MuiTypography-gutterBottom"
        >
          College of university
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          Country: Phillipines
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiCardActions-root makeStyles-cardFooter-4 MuiCardActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Share
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Visit Website
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
